### PR TITLE
Implemented MBC1/MBC2/MBC5 bank switching in transfer pak code

### PR DESCRIPTION
**Pull request by joeldipops**

### CODE REVIEW

**Comment by joeldipops** [comment]
Updated to add mbc5 and mbc2

File: si/gb.c [hunk 1]
```diff
@@ -210,22 +382,302 @@ void mbc_mbc3_write_clock_data_latch( struct controller *controller, uint16_t ad
...
+    cart->extram_size = controller->tpak_save.size;
+  } else {
+    cart->extram = calloc( 1, 65536 );  // TODO: deallocate this eventually?
+    cart->extram_size = 65536;
+  }   
+  cart->extram_bank = cart->extram;
+
+  uint8_t rom_size_code = cart->cartrom_bank_zero[ROM_SIZE_CODE_ADDRESS];
+  if (rom_size_code <= 8) {
+    cart->cartrom_num_banks = power(2, rom_size_code + 1);
```
Comment (sp1187) [inline si/gb.c]: Why do you need the power method? 2<sup>x</sup> can be calculated in a simpler and faster way with just `1<<x`, so `cart->cartrom_num_banks = 1 << (rom_size_code + 1)` would be a better option here.

Comment (joeldipops) [reply si/gb.c]: Thanks for taking a look at it, I have updated the PR with your suggestion.

**Comment by jkbenaim** [comment]
LGTM :+1:

**Pull request merged**